src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (...args) => {
+    if (args.length === 0) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
+      sum += args.reduce((acc, num) => acc + num, 0);
```

Comment:
The 'reduce' method is not needed here because you are only dealing with a single value at a time. You can directly add the argument to 'sum' if it's not undefined.